subsys/usb/usb_device.c

```diff
@@ -593,7 +641,7 @@ static bool usb_set_interface(u8_t iface, u8_t alt_setting)
 			    (cur_alt_setting != alt_setting)) {
 				break;
 			}
-
+			usb_dc_ep_disable(((struct usb_ep_descriptor *)p)->bEndpointAddress);
```

Comment:
Why do we need disable here?

Reply:
@finikorg this is the main point of this PR. There may be a need to disable or remove an endpoint. This functionality is now missing in Zephyr. I wish to discuss how to provide it. This code change is an illustration that actually works.

Reply:
@barsok Don't you have already usb_clear_interface() ?